frame/merkle-mountain-range/rpc/src/lib.rs

```diff
@@ -96,63 +76,59 @@ impl<BlockHash> LeafBatchProof<BlockHash> {
 
 /// MMR RPC methods.
 #[rpc(client, server)]
-pub trait MmrApi<BlockHash, BlockNumber> {
-	/// Generate MMR proof for given block number.
-	///
-	/// This method calls into a runtime with MMR pallet included and attempts to generate
-	/// MMR proof for a block with a specified `block_number`.
-	/// Optionally, a block hash at which the runtime should be queried can be specified.
-	///
-	/// Returns the (full) leaf itself and a proof for this leaf (compact encoding, i.e. hash of
-	/// the leaf). Both parameters are SCALE-encoded.
-	#[method(name = "mmr_generateProof")]
-	fn generate_proof(
-		&self,
-		block_number: BlockNumber,
-		at: Option<BlockHash>,
-	) -> RpcResult<LeafProof<BlockHash>>;
-
-	/// Generate MMR proof for the given block numbers.
-	///
-	/// This method calls into a runtime with MMR pallet included and attempts to generate
-	/// MMR proof for a set of blocks with the specific `block_numbers`.
-	/// Optionally, a block hash at which the runtime should be queried can be specified.
-	///
-	/// Returns the leaves and a proof for these leaves (compact encoding, i.e. hash of
-	/// the leaves). Both parameters are SCALE-encoded.
-	/// The order of entries in the `leaves` field of the returned struct
-	/// is the same as the order of the entries in `block_numbers` supplied
-	#[method(name = "mmr_generateBatchProof")]
-	fn generate_batch_proof(
-		&self,
-		block_numbers: Vec<BlockNumber>,
-		at: Option<BlockHash>,
-	) -> RpcResult<LeafBatchProof<BlockHash>>;
+pub trait MmrApi<BlockHash, BlockNumber, MmrHash> {
+	/// Get the MMR root hash for the current best block.
+	#[method(name = "mmr_root")]
+	fn mmr_root(&self, at: Option<BlockHash>) -> RpcResult<MmrHash>;
 
-	/// Generate a MMR proof for the given `block_numbers` given the `best_known_block_number`.
+	/// Generate an MMR proof for the given `block_numbers`.
```

Comment:
nit: if you want to go the `an MMR` route, you need to change it everywhere :laughing:

Reply:
ack 😄. done in https://github.com/paritytech/substrate/pull/12530/commits/212161d5cddf041060025b8ded705cd266c51546.